solutions/src/test/java/ru/sberbank/school/task06/CollectionUtilsTest.java

```diff
@@ -0,0 +1,214 @@
+        plants.add(new Tree());
+        plants.add(new Maple());
+
+        Oak expectedTarget = new Oak();
+        plants.add(expectedTarget);
+
+        List<Tree> toRemove = new ArrayList<>();
+        Tree tree1 = (Tree) plants.get(0);
+        Maple tree2 = (Maple) plants.get(1);
+        add(toRemove, tree1);
+        add(toRemove, tree2);
+
+        System.out.println("Before remove\n" + plants);
+
+        removeAll(plants, toRemove);
+
+        System.out.println("To remove\n" + toRemove + "\nAfter remove\n" + plants);
+
+        assertEquals(expectedTarget, plants.get(0));
+    }
+
+    @Test
+    public void containsAllTestTrue() {
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Tree());
+        plants.add(new Maple());
+        plants.add(new Oak());
+
+        List<Tree> contains = new ArrayList<>();
+        Tree tree1 = (Tree) plants.get(0);
+        Maple tree2 = (Maple) plants.get(1);
+        add(contains, tree1);
+        add(contains, tree2);
+
+        boolean containsAll = containsAll(plants, contains);
+        assertTrue(containsAll);
+    }
+
+    @Test
+    public void containsAllTestFalse() {
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Tree());
+        plants.add(new Maple());
+        plants.add(new Oak());
+
+        List<Tree> contains = new ArrayList<>();
+        Tree tree1 = (Tree) plants.get(0);
+        Maple tree2 = (Maple) plants.get(1);
+        add(contains, tree1);
+        add(contains, tree2);
+        add(contains, new Oak());
+
+        boolean containsAll = containsAll(plants, contains);
+        assertFalse(containsAll);
+    }
+
+    @Test
+    public void containsAnyTestTrue() {
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Tree());
+        plants.add(new Maple());
+        plants.add(new Oak());
+
+        List<Tree> contains = new ArrayList<>();
+        Tree tree1 = (Tree) plants.get(0);
+        add(contains, tree1);
+        add(contains, new Maple());
+        add(contains, new Oak());
+
+        boolean containsAny = containsAny(plants, contains);
+        assertTrue(containsAny);
+    }
+
+    @Test
+    public void containsAnyTestFalse() {
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Tree());
+        plants.add(new Maple());
+        plants.add(new Oak());
+
+        List<Tree> contains = new ArrayList<>();
+        add(contains, new Maple());
+        add(contains, new Oak());
+
+        boolean containsAny = containsAny(plants, contains);
+        assertFalse(containsAny);
+    }
+
+    @Test
+    public void rangeTestComparable() {
+        List<Integer> target = Arrays.asList(3, 4, 5, 6);
+        List<Integer> list = range(Arrays.asList(8, 1, 3, 5, 6, 4), 3, 6);
+        assertThat(list, equalTo(target));
+    }
+
+    @Test
+    public void rangeTestComparator() {
+        Comparator<Plant> comparator = new Comparator<Plant>() {
```

Comment:
Distance between variable &#39;comparator&#39; declaration and its first usage is 9, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a48119236>*